oci/table_oci_core_nat_gateway.go

```diff
@@ -0,0 +1,242 @@
+	"github.com/oracle/oci-go-sdk/v36/core"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableCoreNatGateway(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_nat_gateway",
+		Description: "OCI Core Nat Gateway",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AnyColumn([]string{"id"}),
+			Hydrate:    getCoreNatGateway,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listCoreNatGateways,
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The OCID of the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name of the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the NAT gateway was created",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The NAT gateway's current state.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "vcn_id",
+				Description: "The OCID of the VCN the NAT gateway belongs to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "nat_ip",
+				Description: "The IP address associated with the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "public_ip_id",
+				Description: "The OCID of the public IP address associated with the NAT gateway.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "block_traffic",
+				Description: "Specifies whether the NAT gateway blocks traffic through it.",
+				Type:        proto.ColumnType_BOOL,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(natGatewayTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
```

Comment:
```| region      | id                                                                                            |
+-------------+-----------------------------------------------------------------------------------------------+
| ap-mumbai-1 | ocid1.natgateway.oc1.ap-mumbai-1.aaaaaaaappey6la4iw4fkem7oos3vd7nkyuir34vx2u4qepmv2d37nwtagjq |
| iad         | ocid1.natgateway.oc1.iad.aaaaaaaakvpa3l5c5i7soxof3rvluz5hy6eclkxe2cx3f3y6zslv5fiedwca         |
+-------------+-----------------------------------------------------------------------------------------------+
```
we are getting region as `iad` instead of `us-ashburn-1`, please check.